### ISSUE TITLE
NullReferenceException with SelectInputGUI.cs

### ISSUE THREAD

**Issue by Darth-Carrotpie** [zombience/audio_analyzer#4]
Basically, in default scene currently there is no GUI script on any object (I mean the one for selecting device input). If I put the script called SelectInputGUI.cs on any object, 2 buttons appear on play, but when clicked each of them spits an error:
"NullReferenceException: Object reference not set to an instance of an object
SelectInputGUI.OnGUI () (at Assets/Scripts/AudioAnalyzer/SelectInputGUI.cs:17)"

**Comment by Darth-Carrotpie** [comment]
P.S. unrelated suggestion: default scene also wont work, since there is "audio missing" from Analyzer script, and default "UseBackedAudio" is ticked On. So my suggestion would be either do the reference properly (with some kind of default example audio file) or set to use mic by default, for audio playback to work right after first git pull.

**Comment by zombience** [comment]
Ah, OK. That may have resulted from a recent fix I made to accommodate an
update that unity made to the audio back end. The issue was that turning
the volume down to zero on an audio source now causes the audio source to
stop processing audio at all. This is likely an efficiency update on
unity's part.

I'm not currently at a computer, so I can't confirm, but I suspect either
my update or unity's update changed something that affects the way I had
been selecting inputs.

I can likely fix the gui issue today, although not until a few hours from
now. But the audio gap issue will have to wait until I can secure a larger
block of time.

Please open an issue for the gui bug and I will address it later today.

Cheers

Jason Araujo

---

Not sure why it's important to note that this was sent from a phone... But
it was.

**Comment by zombience** [comment]
Oh woops, you did open a separate issue.
Heh! I'll take care of it today.

Jason Araujo

---

Not sure why it's important to note that this was sent from a phone... But
it was.

**Comment by Darth-Carrotpie** [comment]
Yeah... it was me who pinged you about the volume issue too 😄 
Sorry for the extra work... Your analyzer is great! ;)

**Comment by zombience** [comment]
Glad you find it useful!
I'm grateful for having bugs pointed out. This repo is falling behind
unity's updates bc I've been away from a computer for months and haven't
used it for any projects in a while.

Jason Araujo

---

Not sure why it's important to note that this was sent from a phone... But
it was.

**Comment by zombience** [comment]
Hey there, 
so this is a case of bad (e.g. none) documentation on my part. 
You do not need to manually add the SelectInputGUI to the scene. It is called within the AudioAnalyzer code. If you hit the key "I", it will automatically add the necessary component, display the selections available, and then when you hit "I" again, the component will be removed, and your selection will be stored. 

I manually added the component to a scene object and achieved the behavior described in this issue. 
I am going to do two things: 
1) put demo audio in so that baked audio can be used, and set that to default. 
2) close this issue and open another issue citing lack of documentation 

I may or may not have time to get to documentation today, but I should be able to do it this week. 

Hope that helps

Cheers
-j
